apps/account-worker/src/transaction_publisher/publisher.service.ts

```diff
@@ -149,6 +155,29 @@ export class TransactionPublisherService extends BaseConsumer implements OnAppli
     }
   }
 
+  /**
+   * Determine if job should be returned to the queue for retries.
+   * (Default retry logic is "1"; ie only a single attempt. But some chain
+   * submission errors deserve a retry; however, we don't want to retry indefinitely,
+   * so we cap it at 4)
+   * @param job The current job
+   * @param err The error that was thrown
+   * @returns Whether the job should be retried
+   */
+  public static shouldRetry(job: Job<any>, err: Error): boolean {
+    if (
+      job.attemptsStarted < 4 &&
+      (err instanceof NonceConflictError ||
+        err instanceof BadSignatureError ||
```

Comment:
If we get the signature from the user then retrying a bad signature would not work but that might not be the case in here

Reply:
I believe that is a different error

Reply:
The bad signature is the provider submitting the transaction, not the payload inside the transaction... that would, I think, be an extrinsic error on-chain.